docs/ContainerRuntimeConfigDesign.md

```diff
@@ -0,0 +1,95 @@
+# Summary
+
+Users need a way to update the container runtime configuration. Users can do this via MCO currently, but they need to know the correct options to use. We can hopefully help customers by documenting a CRD with the ContainerRuntime tuneables and have the MCO use these tuneables when rendering the crio.conf, storage.conf, and registries.conf files to Ignition/disk.
+
+The CRI-O runtime is used by default and is the recommended solution.
+
+# Motivation
+
+The MCO contains the logic to write the crio.conf, storage.conf, and registries.conf configuration files to Ignition. When Ignition starts on a machine it writes these three files to configure the container runtime. When these resources change within the MachineConfig, the new configs are rewritten and the MachineConfigDaemon is instructed to reboot the machine to pick up the new configs.
+
+## Goals
+
+The following goals should not require explicit knowledge of a customer to know the right command-line or configuration options to use within the ContainerRuntime.
+
+1. setting pids limit
```

Comment:
What happens if this becomes a kube level setting in the future?

Reply:
kubelet settings apply at the pod level while this setting is at the container level. We can advise admins to not set this or keep this unbounded once kubelet setting lands.

Reply:
Kubelet setting is pod level.

We backported the flag to set a default per pod pod limit to 3.11+ from 1.14 to address production issues.

Robert is working on enable node to pod pod isolation.

SIG node kep I wrote proposes a granular pod limit feature as a potential policy knob in future but at pod and not Container boundary.  If we do go to Container boundary in kubelet, crio value is default absent kubelet telling it a specific value.